### DISCUSSION TITLE
Internal and extrenal states

### DISCUSSION THREAD

**Discussion by Wonder1905** [lucidrains/pi-zero-pytorch#7] [Q&A]
Hi, in the code there are :
"external_state_tokens"
"internal_state_tokens"
What are they for?

Thanks!

**Comment by lucidrains** [comment]
@Wonder1905 hey again (noticed you are also on transfusion repo)

that is for extending the system to incorporate more states than just one joint state token (for internal), and a few images (for external)

**Reply by Wonder1905** [reply]
Yes, I'm fascinated from this architecture family.
Got it! thanks!

**Reply by lucidrains** [reply]
yea, it is very new

thankfully, [flex attention](https://pytorch.org/blog/flexattention/) came along and makes it very accessible
